tavern/tomes/process_list/main.eldritch

```diff
@@ -8,19 +8,32 @@ def pad_username (username):
     padding = username_column_width - len(username)
     return username +  " "*padding
 
+
+def print_table(rows: list[list]):
+    """Pretty print a table, auto adjusting width based on the row data"""
+    def rpad(s: str, n: int, c=" "):
+        return s + (n-len(s))*c
+    # count columns
+    counts = [0]*len(rows[0])
+    for r in rows:
+        for i, f in enumerate(r):
+            counts[i] = max(len(f), counts[i])
+    # Print the columns
+    for r in rows:
+        row = []
+        for i, c in enumerate(r):
+            row.append(rpad(c, counts[i]+2))
+        print("".join(row))
+
 def process_list(cmd_substring):
 
     if cmd_substring == '*':
         cmd_substring = ''
 
     procs = process.list()
 
-    print(pad_pid("PID"))
-    print(pad_pid("PPID")) 
-    print(pad_username("username"))
-    print("command\n")
-
-    for proc in procs:
+    rows = [["PID", "PPID", "USER","COMMAND"]]
+    for proc in sorted(procs, key=lambda x:x["pid"]):
```

Comment:
Sorting by PID? smh - if possible we should sort by spawn time (which i think is the default unsure thogh)